ui/app/common/displaycontrols/observation/services/formRecordTreeBuildService.js

```diff
@@ -148,10 +155,16 @@ angular.module('bahmni.common.displaycontrol.observation')
                         "conceptClass": null
                     }
                 };
-                obsGroup.concept.shortName = column.value;
+                var translationKey = column.translationKey;
+                if (translationData && translationData.labels && translationData.labels[translationKey][0] !== translationKey) {
+                    obsGroup.concept.shortName = translationData.labels[translationKey][0];
+                }
```

Comment:
Hi @vinishad, can you pull this (L158-L164) into a method so that it's more readable

Reply:
@sumanmaity112 thanks for the comments. Will work on that.